packages/components/src/navigator/navigator-provider/component.tsx

```diff
@@ -47,7 +91,7 @@ function NavigatorProvider(
 
 	return (
 		<View ref={ forwardedRef } className={ classes } { ...otherProps }>
-			<NavigatorContext.Provider value={ [ path, setPath ] }>
+			<NavigatorContext.Provider value={ navigatorContextValue }>
```

Comment:
The previous values of the state (path, setPath) were not new instances on each render, while now they are, it might not matter because the whole value was a new instance but it could have an impact depending on they are consumed. Should we useCallback on the push, pop?

Reply:
That's a good point, I'll look into it

Reply:
> Should we useCallback on the push, pop?

Done in https://github.com/WordPress/gutenberg/pull/37223/commits/8868acf9bd3f9b1ca522b3b5fca8a6a7452dfe2f